src/tsickle.ts

```diff
@@ -867,7 +867,8 @@ class Annotator extends ClosureRewriter {
         // If it has a symbol, it's actually a regular declared property.
         if (!quotedPropSym) return false;
         const declarationHasQuotes =
-            !quotedPropSym.declarations || quotedPropSym.declarations.some(decl => {
+            !quotedPropSym.declarations || quotedPropSym.declarations.some(d => {
+              const decl = d as ts.NamedDeclaration;
               if (!decl.name) return false;
```

Comment:
The definition is:
```
    interface NamedDeclaration extends Declaration {
        name?: DeclarationName;
    }
```
so it's always safe to coerce downward and test the attribute.

It looks like upstream they're trying to iteratively migrate code to using NamedDeclaration where needed: https://github.com/Microsoft/TypeScript/commit/9392342f7807c28738199ce82c73e52ac3bc73ef